wagtail/core/models.py

```diff
@@ -42,6 +42,120 @@
 PAGE_TEMPLATE_VAR = 'page'
 
 
+class MultiTableCopyMixin:
+    default_exclude_fields_in_copy = ['id']
+
+    def _get_field_dictionaries(self, exclude_fields=None, **kwargs):
+        specific_self = self.specific
+        exclude_fields = exclude_fields or []
+        specific_dict = {}
+        specific_m2m_dict = {}
+
+        for field in specific_self._meta.get_fields():
+            # Ignore explicitly excluded fields
+            if field.name in exclude_fields:
+                continue
+
+            # Ignore reverse relations
+            if field.auto_created:
+                continue
+
+            # Copy child m2m relations
+            # Otherwise add them to the m2m dict
+            if field.many_to_many:
+                if isinstance(field, ParentalManyToManyField):
+                    parental_field = getattr(specific_self, field.name)
+                    if hasattr(parental_field, 'all'):
+                        values = parental_field.all()
+                        if values:
+                            specific_dict[field.name] = values
+                else:
+                    specific_m2m_dict[field.name] = getattr(specific_self, field.name).all()
+                continue
+
+            # Ignore parent links (page_ptr)
+            if isinstance(field, models.OneToOneField) and field.remote_field.parent_link:
+                continue
+
+            specific_dict[field.name] = getattr(specific_self, field.name)
+
+        return specific_dict, specific_m2m_dict
+
+    def _get_copy_instance(self, specific_dict, specific_m2m_dict, update_attrs=None, **kwargs):
+        if not update_attrs:
+            update_attrs = {}
+
+        specific_class = self.specific.__class__
+
+        copy_instance = specific_class(**specific_dict)
+
+        if update_attrs:
+            for field, value in update_attrs.items():
+                if field in specific_m2m_dict:
+                    continue
+                setattr(copy_instance, field, value)
+
+        return copy_instance
+
+    def _save_copy_instance(self, instance, **kwargs):
+        raise NotImplementedError
+
+    def _set_m2m_relations(self, instance, specific_m2m_dict, update_attrs=None, **kwargs):
+        if not update_attrs:
+            update_attrs = {}
+        for field_name, value in specific_m2m_dict.items():
+            value = update_attrs.get(field_name, value)
+            getattr(instance, field_name).set(value)
+
+        return self._save_copy_instance(instance)
```

Comment:
If this method is no longer handling ParentalManyToMany relations (as per a7612974fc4aa7dc7220ae21c606ab811c9a5d80), then I think this _save_copy_instance call can be removed - and then Page._save_copy_instance won't have to do the `if not instance.id` thing.

Reply:
Thanks, I've removed it now (as well as the `instance.id` check)